### DISCUSSION TITLE
How to return request Id immediately?

### DISCUSSION THREAD

**Discussion by RobTF9** [deepgram/community#109] [General help] [answered]
I'm trying to use the callback option to get a request id and then recieve the transcription response via a webhook.

Only problem is that it takes the same amount of time to recieve the request id as it does the actual transription via webhook.

Using the latest version of the node sdk, this is my code:

```
import { Deepgram } from '@deepgram/sdk'
import config from '../config'
import fs from 'fs'

const deepgram = new Deepgram(config.deepgram || '')

export async function transcribeVideo(file: Express.Multer.File) {
  const audioSource = {
    stream: fs.createReadStream(file.path),
    mimetype: file.mimetype,
  }

  const response = await deepgram.transcription.preRecorded(audioSource, {
    punctuate: true,
    diarize: true,
    tier: 'enhanced',
    callback: config.notification_url,
  })

  return response
}
```

**Comment by briancbarrow** [comment]
Hi @RobTF9, Right now the request id is part of the standard response we send. Can I ask what your use case is where you need the request id before the response? That might help me understand it to find a solution for you.

**Reply by RobTF9** [reply]
I'd like to get the request id before the transcription is finished, that way I can return a response to the user that their video upload was succesful and their transcription is processing. Once the transcription is finished (and rest of deepgram response recieved via webhook) I can update the database and send a websocket event to the client.

Does that make sense? Alternatively would be good if I could send the id I'm using in my db as the request id

**Reply by briancbarrow** [reply, marked as answer]
I think it makes sense. The callback option sends the transcription response to the callback URL once Deepgram is done processing it. When using the callback option, the initial call to Deepgram will return with the request_id in the response body. So you shouldn't need the callback in order to get the request_id. You should be able to get it before the callback url is called with the transcription response.

Does that help?
